bridged-tokens-wrapper/src/lib.rs

```diff
@@ -132,23 +186,27 @@ pub trait BridgedTokensWrapper {
             payment_token == universal_bridged_token_ids,
             "Esdt token unavailable"
         );
-
+        self.require_tokens_have_set_decimals_num(&payment_token, &requested_token);
         let chain_specific_token_id = &requested_token;
+        let universal_token_decimals = self.token_decimals_num(&payment_token).get();
```

Comment:
you could make an internal function for the conversion and call the same function on both wrap and unwrap functions.